pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -2720,25 +2746,55 @@ private void updatePolicies(NamespaceName ns, Function<Policies, Policies> updat
-                throw new RestException(e);
-            }
-        }
 
-        internalSetPolicies("resource_group_name", rgName);
+    protected void internalGetNamespaceResourceGroup(AsyncResponse asyncResponse, String tenant, String namespace) {
+        validateNamespacePolicyOperationAsync(NamespaceName.get(tenant, namespace),
+                PolicyName.RESOURCEGROUP, PolicyOperation.READ)
+                .thenCompose(__ -> getNamespacePoliciesAsync(namespaceName)
+                        .thenAccept(policies -> asyncResponse.resume(policies.resource_group_name)))
+                .exceptionally(ex -> {
+                    Throwable realCause = FutureUtil.unwrapCompletionException(ex);
+                    if (realCause instanceof WebApplicationException) {
+                        log.info("[{}] Successfully to get namespace resource group {}/{}",
+                                clientAppId(), tenant, namespace);
+                        asyncResponse.resume(realCause);
+                    } else {
+                        log.error("[{}] Fail to get namespace resource group {}/{}", clientAppId(), tenant, namespace);
+                        asyncResponse.resume(new RestException(realCause));
+                    }
+                    return null;
+                });
     }
 
-
-    private static final Logger log = LoggerFactory.getLogger(NamespacesBase.class);
+    protected void internalSetNamespaceResourceGroup(AsyncResponse asyncResponse, String rgName) {
+        validateNamespacePolicyOperationAsync(namespaceName, PolicyName.RESOURCEGROUP, PolicyOperation.WRITE)
+                .thenCompose(__ -> validatePoliciesReadOnlyAccessAsync())
+                .thenCompose(__ -> {
+                    if (rgName != null) {
+                        return resourceGroupResources().getResourceGroupAsync(rgName)
+                                .thenAccept(resourceGroup -> {
+                                    // check resource group exists.
+                                    if (!resourceGroup.isPresent()) {
+                                        throw new RestException(Status.PRECONDITION_FAILED,
+                                                "ResourceGroup does not exist");
+                                    }
+                                });
+                    } else {
+                        return CompletableFuture.completedFuture(null);
+                    }
+                }).thenCompose(__ -> internalSetPoliciesAsync("resource_group_name", rgName))
+                .thenAccept(__ -> {
+                    log.info("[{}] Successfully to set namespace resource group {}", clientAppId(), rgName);
+                    asyncResponse.resume(Response.noContent().build());
+                }).exceptionally(ex -> {
+                    Throwable realCause = FutureUtil.unwrapCompletionException(ex);
+                    if (realCause instanceof WebApplicationException) {
```

Comment:
```suggestion
log.error("[{}] Fail to set namespace resource group {}: {}", clientAppId(), rgName, realCause);
resumeAsyncResponseExceptionally(asyncResponse, realCause);
return null;
```

Reply:
I think the exception message is enough.